bottle_app.py

```diff
@@ -475,4 +477,4 @@ def app():
         if not os.path.exists(args.dbcredentials):
             raise FileNotFoundError(args.dbcredentials)
         paths.BOTTLE_APP_INI = args.dbcredentials
-    bottle.default_app().run(host='localhost',debug=True, reloader=True)
+    bottle.default_app().run(host='localhost',debug=True, reloader=True, port=args.port)
```

Comment:
Because bottle.run can't deal with port=None, this line makes it necessary to explicitly specify a --port argument on the command line else we get:

TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NoneType'
Traceback (most recent call last):
  File "/Users/sbarber/Library/Python/3.11/lib/python/site-packages/bottle.py", line 3157, in run
    server = server(host=host, port=port, **kargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sbarber/Library/Python/3.11/lib/python/site-packages/bottle.py", line 2753, in __init__
    self.port = int(port)
                ^^^^^^^^^

Would be good, I think, to let --port default when not specified.

Reply:
Fixed in b71a02c, along with some other damage. (I had python3.9 hard-coded in my path on my Mac, so I wasn't seeing the other dependency problems.)